### PR TITLE
Print stderr from go tool pprof in profile gatherer

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: test/e2e/framework/profile_gatherer.go [hunk 1]
```diff
@@ -107,8 +108,10 @@ func gatherProfileOfKind(profileBaseName, kind string) error {
 	}
 	defer outfile.Close()
 	cmd.Stdout = outfile
+	stderr := bytes.NewBuffer(nil)
+	cmd.Stderr = stderr
```
Comment (shyamjvs) [inline test/e2e/framework/profile_gatherer.go]: This is cleaner IMO:

```
cmd.Stderr = &errbuf
return fmt.Errorf("Failed to run 'go tool pprof': %v, stderr: %v", err, errbuf.String())
```

Comment (shyamjvs) [reply test/e2e/framework/profile_gatherer.go]: Sorry, the `&` there is not required. But the 2nd line still holds.

Comment (wojtek-t) [reply test/e2e/framework/profile_gatherer.go]: done

**Comment by shyamjvs** [comment]
Ref umbrella issue - https://github.com/kubernetes/kubernetes/issues/59760

**Pull request merged**